### DISCUSSION TITLE
Compound Components

### DISCUSSION THREAD

**Discussion by joshpalmeroc** [storybookjs/storybook#25552] [Help] [answered]
## Framework: Next JS 14

I would like to have a story that implements the compound component pattern in react. I have tried using decorators, render function, subcomponents, and followed these documentation links https://storybook.js.org/docs/writing-stories/stories-for-multiple-components. It seems like whatever code I try gives me an error. Im curious if anyone else has tried to implement this pattern into their stories.

### Additional information

```js
import type { Meta, StoryObj } from "@storybook/react";
import {
    Accordion,
    AccordionItem,
    AccordionContent,
    AccordionTrigger,
} from "@dev/ui";

const meta = {
    title: "Components/Accordion",
    component: Accordion,
    tags: ["autodocs"],
} satisfies Meta<typeof Accordion>;

export default meta;

type Story = StoryObj<typeof meta>;

// // More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
// export const Primary = {};
export const Primary = () => {
	return (
		<Accordion>
			<AccordionItem>
				<AccordionTrigger>Trigger</AccordionTrigger>
				<AccordionContent>Content goes here</AccordionContent>
			</AccordionItem>
			<AccordionItem>
				<AccordionTrigger>Trigger</AccordionTrigger>
				<AccordionContent>Content goes here</AccordionContent>
			</AccordionItem>
		</Accordion>
	);
};

```

**Comment by joshpalmeroc** [comment, marked as answer]
I realized I needed to make my file a .tsx file rather then a .ts file. I then used the render method and it worked

**Reply by omerzirh** [reply]
I spend hours to solve same issue and found your comment. Thanks
